htmresearch/algorithms/apical_dependent_temporal_memory.py

```diff
@@ -386,9 +408,16 @@ def _calculateLearning(self,
 
   @staticmethod
   def _calculateSegmentActivity(connections, activeInput, connectedPermanence,
-                                activationThreshold, minThreshold):
+                                activationThreshold, minThreshold,
+                                reducedBasalThreshold,
+                                reducedBasalThresholdCells = ()):
```

Comment:
These could just be `reducedThreshold` and `reducedThresholdCells`. This method should never mention "apical" or "basal" -- it can be used for both. If someone wanted, they could use this method as-is to reduce the apical threshold for a set of cells.